clients/include/testing_trsm_strided_batched.hpp

```diff
@@ -0,0 +1,439 @@
+/* ************************************************************************
+ * Copyright 2018-2019 Advanced Micro Devices, Inc.
+ * ************************************************************************ */
+
+#include "cblas_interface.hpp"
+#include "flops.hpp"
+#include "norm.hpp"
+#include "rocblas.hpp"
+#include "rocblas_datatype2string.hpp"
+#include "rocblas_init.hpp"
+#include "rocblas_math.hpp"
+#include "rocblas_random.hpp"
+#include "rocblas_test.hpp"
+#include "rocblas_vector.hpp"
+#include "unit.hpp"
+#include "utility.hpp"
+
+#define ERROR_EPS_MULTIPLIER 40
+#define RESIDUAL_EPS_MULTIPLIER 20
+
+template <typename T>
+void testing_trsm_strided_batched(const Arguments& arg)
+{
+    rocblas_int M           = arg.M;
+    rocblas_int N           = arg.N;
+    rocblas_int lda         = arg.lda;
+    rocblas_int ldb         = arg.ldb;
+    rocblas_int stride_a    = arg.stride_a;
+    rocblas_int stride_b    = arg.stride_b;
+    rocblas_int batch_count = arg.batch_count;
+
+    char char_side   = arg.side;
+    char char_uplo   = arg.uplo;
+    char char_transA = arg.transA;
+    char char_diag   = arg.diag;
+    T    alpha_h     = arg.alpha;
+
+    rocblas_side      side   = char2rocblas_side(char_side);
+    rocblas_fill      uplo   = char2rocblas_fill(char_uplo);
+    rocblas_operation transA = char2rocblas_operation(char_transA);
+    rocblas_diagonal  diag   = char2rocblas_diagonal(char_diag);
+
+    rocblas_int K      = side == rocblas_side_left ? M : N;
+    size_t      size_A = lda * size_t(K) * batch_count + size_t(stride_a) * size_t(batch_count - 1);
```

Comment:
correct if I am wrong but I think this should be:
lda * size_t(K) + size_t(stride_a) * size_t(batch_count - 1);

Reply:
> 
> 
> correct if I am wrong but I think this should be:
> lda * size_t(K) + size_t(stride_a) * size_t(batch_count - 1);
```
lda * size_t(K) + stride_a * (batch_count -1)
```
There is no need to cast `stride_a` or any other `rocblas_stride` variables, since they are supposed to already be `ptrdiff_t` size. `batch_count - 1`  does not need to be cast because it will be automatically promoted.

Reply:
Good catch, fixed.